Robust.Client/UserInterface/UserInterfaceManager.cs

```diff
@@ -954,11 +956,33 @@ private void WindowContentScaleChanged(WindowContentScaleEventArgs args)
                 UpdateUIScale(root);
         }
 
-        private void UpdateUIScale(WindowRoot root)
+        private float CalculateAutoScale(WindowRoot root)
         {
-            var newVal = _configurationManager.GetCVar(CVars.DisplayUIScale);
-            root.UIScaleSet = newVal == 0f ? root.Window.ContentScale.X : newVal;
+            //Grab the OS UIScale or the value set through CVAR debug
+            var osScale = _configurationManager.GetCVar(CVars.DisplayUIScale);
+            osScale = osScale == 0f ? root.Window.ContentScale.X : osScale;
+            var windowSize = root.Window.RenderTarget.Size;
+            //Only run autoscale if it is enabled, otherwise default to just use OS UIScale
+            if (!root.AutoScale && (windowSize.X <= 0 || windowSize.Y <= 0)) return osScale;
+            var maxScaleRes = root.AutoScaleUpperCutoff;
+            var minScaleRes = root.AutoScaleLowerCutoff;
+            var autoScaleMin = root.AutoScaleMinimum;
+            float scaleRatioX;
+            float scaleRatioY;
+            
+            //Calculate the scale ratios and clamp it between the maximums and minimums
+            scaleRatioX = (windowSize.X > maxScaleRes.X) ? osScale : ((float)windowSize.X-minScaleRes.X)/(maxScaleRes.X-minScaleRes.X)*osScale;
+            scaleRatioX = (scaleRatioX > autoScaleMin) ? scaleRatioX : autoScaleMin;
```

Comment:
i think you can condense this into
```suggestion
            scaleRatioX = Math.Clamp(((float)windowSize.X-minScaleRes.X)/(maxScaleRes.X-minScaleRes.X)*osScale, autoScaleMin, osScale)
```